internal/project/project.go

```diff
@@ -1,241 +1,232 @@
+	data, err := p.ReadMarkdownFile(relativePath, nil)
+	if err != nil {
+		return nil, err
 	}
-	cmd := exec.Command(cmdSlice[0], cmdSlice[1:]...)
-	cmd.Dir = cwd
-	out, err := cmd.CombinedOutput()
+
+	doc := document.New(data, cmark.Render)
+	node, _, err := doc.Parse()
 	if err != nil {
 		return nil, err
 	}
 
-	stdout := string(out)
-	branches := getBranchNamesFromStdout(stdout, greedy)
-	return branches, nil
-}
+	blocks := document.CollectCodeBlocks(node)
 
-func getBranchNamesFromStdout(stdout string, greedy bool) []Branch {
-	var branches []Branch
-	modifier := "(?mU)"
-	if greedy {
-		modifier = "(?m)"
-	}
-	re := regexp.MustCompile(fmt.Sprintf(`%s(\S+)[:\/](\S+)\s?$`, modifier))
-	for _, line := range strings.Split(stdout, "\n") {
-		split := strings.Split(line, "--||--")
-		orgbranch := re.FindStringSubmatch(split[0])
-		if len(orgbranch) == 3 && len(split) > 1 && len(split[1]) > 1 {
-			branches = append(branches, Branch{Name: orgbranch[2], Description: split[1]})
+	filtered := make(document.CodeBlocks, 0, len(blocks))
+	for _, b := range blocks {
+		// ignore blocks without name if ignoreNameless param is set
+		if ignoreNameless && b.Attributes()["name"] == "" {
+			continue
+		}
+
+		if allowUnknown || (b.Language() != "" && runner.IsSupported(b.Language())) {
+			filtered = append(filtered, b)
 		}
 	}
-	return branches
+
+	return filtered, nil
 }
 
-func GetUsersBranches(repoUser string) ([]Branch, error) {
-	var email string
-	cwd, err := os.Getwd()
+type FileCodeBlocks struct {
+	FileName   string
+	CodeBlocks document.CodeBlocks
+}
+type CodeBlocks []*FileCodeBlocks
+
+func (p *Project) GetAllCodeBlocks(allowUnknown bool, ignoreNameless bool) (CodeBlocks, error) {
+	files, err := p.getAllMarkdownFiles()
 	if err != nil {
 		return nil, err
 	}
 
-	if len(repoUser) > 0 {
-		email = repoUser
-	} else {
-		email, err = GetCurrentGitEmail(cwd)
+	blocks := CodeBlocks{}
+	for _, file := range files {
+		codeBlock, err := p.GetCodeBlocks(file[len(p.RootDir):], allowUnknown, ignoreNameless)
 		if err != nil {
-			return nil, errors.New("could not find current git user")
+			return nil, err
 		}
-	}
 
-	branches, err := GetUsersBranchNames(cwd, email)
-	if err != nil {
-		return nil, errors.New("error while querying user's branches")
+		if len(codeBlock) == 0 {
+			continue
+		}
+
+		blocks = append(blocks, &FileCodeBlocks{
+			FileName:   file[len(p.RootDir):],
+			CodeBlocks: codeBlock,
+		})
 	}
 
-	return branches, nil
+	return blocks, nil
 }
 
-func GetRepoBranches() ([]Branch, error) {
-	cwd, err := os.Getwd()
+func (p *Project) LookUpCodeBlockByID(id string) (*string, *document.CodeBlock, error) {
+	files, err := p.getAllMarkdownFiles()
 	if err != nil {
-		return nil, err
+		return nil, nil, err
 	}
 
-	branches, err := GetBranchNames(cwd)
-	if err != nil {
-		return nil, errors.New("error while querying repository branches")
+	for _, file := range files {
+		codeBlock, err := p.GetCodeBlocks(file[len(p.RootDir):], false, true)
```

Comment:
```suggestion
		codeBlock, err := p.GetCodeBlocks(file[len(p.RootDir):], allowUnknown, ignoreNameless)
```

Should pass these down